app/src/main/java/fr/free/nrw/commons/depictions/Media/DepictedImagesFragment.java

```
@@ -0,0 +1,264 @@
+        return gridAdapter;
+    }
+
+    /**
+     * Initializes the adapter with a list of Media objects
+     *
+     * @param mediaList List of new Media to be displayed
+     */
+    @Override
+    public void setAdapter(List<Media> mediaList) {
+        gridAdapter = new fr.free.nrw.commons.depictions.GridViewAdapter(getContext(), R.layout.layout_depict_image, mediaList);
+        gridView.setAdapter(gridAdapter);
+    }
+
+    /**
+     * adds list to adapter
+     */
+    @Override
+    public void addItemsToAdapter(List<Media> media) {
+        gridAdapter.addAll(media);
+        gridAdapter.notifyDataSetChanged();
+    }
+
+    /**
+     * Sets loading status depending on the boolean value
+     */
+    @Override
+    public void setLoadingStatus(Boolean value) {
+        if (!value) {
+            statusTextView.setVisibility(GONE);
+        }
+        isLoading = value;
+    }
+
+    /**
+     * Inform the view that there are no more items to be loaded for this search query
+     * or reset the isLastPage for the current query
+     * @param isLastPage
+     */
+    @Override
+    public void setIsLastPage(boolean isLastPage) {
+        this.isLastPage=isLastPage;
+        progressBar.setVisibility(GONE);
+    }
+
+
+    /**
+     * Handles the success scenario
+     * On first load, it initializes the grid view. On subsequent loads, it adds items to the adapter
+     *
+     * @param collection List of new Media to be displayed
+     */
+    @Override
+    public void handleSuccess(List<Media> collection) {
+       presenter.addItemsToQueryList(collection);
+        if (gridAdapter == null) {
+            setAdapter(collection);
+        } else {
+            if (gridAdapter.containsAll(collection)) {
+                return;
+            }
+            gridAdapter.addItems(collection);
+
+            try {
+                ((WikidataItemDetailsActivity) getContext()).viewPagerNotifyDataSetChanged();
+            } catch (Exception e) {
```

Why are we trying to catch all exceptions here?

---

I would guess because there was an example in the code before me moved to casting to that interface, the broken window effect.